src/java/org/apache/cassandra/streaming/StreamSession.java

```diff
@@ -694,8 +701,7 @@ public Future<?> onError(Throwable e)
             state(State.FAILED); // make sure subsequent error handling sees the session in a final state 
             channel.sendControlMessage(new SessionFailedMessage()).awaitUninterruptibly();
         }
-
-        return closeSession(State.FAILED);
+        return closeSession(State.FAILED, "Failed because of an unkown exception\n" + Throwables.getStackTraceAsString(e).substring(0,162)); //bounding the stracktrace to propaerly manage memory
```

Comment:
162 is a magic number without any justification, can you flesh this out?  Why was this chosen?